pkg/loader/model.go

```diff
@@ -250,6 +300,62 @@ func (dml *DML) deleteSQL() (sql string, args []interface{}) {
 	return
 }
 
+func (dml *DML) oracleDeleteSQL() (sql string) {
+	builder := new(strings.Builder)
+
+	fmt.Fprintf(builder, "DELETE FROM %s WHERE ", dml.OracleTableName())
+	dml.buildOracleWhere(builder)
+	builder.WriteString(" AND rownum <=1")
+	sql = builder.String()
+	return
+}
+
+func (dml * DML) oracleDeleteNewValueSQL() (sql string) {
+	builder := new(strings.Builder)
+	fmt.Fprintf(builder, "DELETE FROM %s WHERE ", dml.OracleTableName())
+
+	valueMap := dml.Values
+	colNames := make([]string, 0)
+	colValues := make([]interface{}, 0)
+	// Try to use unique key values when available
+	for _, index := range dml.info.uniqueKeys {
+		notAnyNil := true
+		for _, colName := range index.columns {
+			if valueMap[colName] == nil {
+				notAnyNil = false
+				break
+			}
+			colNames = append(colNames, colName)
+			colValues = append(colValues, valueMap[colName])
+		}
+		if !notAnyNil {
+			colNames = colNames[:0]
+			colValues = colValues[:0]
+		}
+	}
+	// Fallback to use all columns
+	if len(colNames) == 0 {
+		for _, col := range dml.columnNames() {
+			colNames = append(colNames, col)
+			colValues = append(colValues, valueMap[col])
+		}
+	}
```

Comment:
Why not use `whereSlice` here? This for loop will return **ALL** unique index columns.

Reply:
it is not suitable for my case